app/models/event_schedule.rb

```diff
@@ -52,4 +54,26 @@ def start_before_start_hour
   def conference_id
     schedule.program.conference_id
   end
+
+  ##
+  # Validates that the event is scheduled in the same room as it's track
+  #
+  def room_of_track
+    if event && event.track.try(:room) && event.track.room != room
+      errors.add(:room, "must be the same as the track's room (#{event.track.room.name})")
+    end
+  end
+
+  ##
+  # Validates that the event is scheduled within it's track's time slot
```

Comment:
Yeah, I know. My hand just adds the apostrophe by itself, when I want to express possession and I'm not really paying attention to what I'm writing :P